packages/frontend/src/pages/SqlRunnerNew.tsx

```diff
@@ -47,13 +51,32 @@ const SqlRunnerNew = ({
     const mode = useAppSelector((state) => state.sqlRunner.mode);
 
     const params = useParams<{ projectUuid: string; slug?: string }>();
+    const share = useSearchParams('share');
+    const { data: sqlRunnerState, error: shareError } = useGetShare(
+        share || undefined,
+    );
 
     const location = useLocation<{ sql?: string }>();
     const history = useHistory();
 
     const [isLeftSidebarOpen, setLeftSidebarOpen] = useState(true);
     const { data: project } = useProject(projectUuid);
 
+    useEffect(() => {
+        if (shareError) return;
```

Comment:
Should we toast the error here? I don't know what might come up, but I imagine mismatched permissions will be a pretty common issue. In which case it would be helpful for people to know.